apis/gateway/v1alpha1/filter_types.go

```diff
@@ -0,0 +1,47 @@
+// Package v1alpha1 contains custom resource definitions for use with the Gateway API.
+//
+//nolint:lll
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PolicyFilter represents a Pomerium policy that can be attached to a particular route defined
+// via the Kubernetes Gateway API.
+//
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type PolicyFilter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the content of the policy.
+	Spec PolicyFilterSpec `json:"spec,omitempty"`
+
+	// +kubebuilder:default={conditions: {{type: "Valid", status: "Unknown", reason:"Pending", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}}
+	Status PolicyFilterStatus `json:"status,omitempty"`
+}
+
+// PolicyFilterSpec defines policy rules.
+type PolicyFilterSpec struct {
+	// Policy rules in Pomerium Policy Language (PPL) syntax. May be expressed
+	// in either YAML or JSON format.
+	PPL string `json:"ppl,omitempty"`
+}
+
+// PolicyFilterStatus represents the state of a PolicyFilter.
+type PolicyFilterStatus struct {
+	// Conditions describe the current state of the PolicyFilter.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=8
```

Comment:
max items validation might not be necessary here

Reply:
It looked like all the Gateway API objects include MaxItems on their status conditions so I thought it made sense to do likewise:
- https://github.com/kubernetes-sigs/gateway-api/blob/1fe2b9f8ee99a6475a65eedd1ce060f363a8634d/apis/v1/gatewayclass_types.go#L259
- https://github.com/kubernetes-sigs/gateway-api/blob/1fe2b9f8ee99a6475a65eedd1ce060f363a8634d/apis/v1/gateway_types.go#L691
- https://github.com/kubernetes-sigs/gateway-api/blob/1fe2b9f8ee99a6475a65eedd1ce060f363a8634d/apis/v1/shared_types.go#L479

Happy to remove it if you think it doesn't make sense.

Reply:
Likely specific to their usage, probably doesn't make sense here unless we have a specific need for it

Reply:
OK, I'll remove it.